test/Microsoft.AspNetCore.Http.Tests/Features/QueryFeatureTests.cs

```diff
@@ -24,5 +24,39 @@ public void QueryReturnsParsedQueryCollection()
             // Assert
             Assert.Equal("bar", queryCollection["foo"]);
         }
+
+        [Theory]
+        [InlineData("?q", "q")]
+        [InlineData("?q&", "q")]
+        [InlineData("?q1=abc&q2", "q2")]
+        public void KeyWithoutValuesAddedToQueryCollection(string queryString, string emptyParam)
+        {
+            var features = new FeatureCollection();
+            var request = new HttpRequestFeature();
+            request.QueryString = queryString;
+            features[typeof(IHttpRequestFeature)] = request;
+
+            var provider = new QueryFeature(features);
+
+            var queryCollection = provider.Query;
+
+            Assert.True(queryCollection.Keys.Contains(emptyParam));
+            Assert.Equal(string.Empty, queryCollection[emptyParam]);
+        }
+
+        [Fact]
+        public void EmptyKeysNotAddedToQueryCollection()
+        {
+            var features = new FeatureCollection();
+            var request = new HttpRequestFeature();
+            request.QueryString = "&&";
```

Comment:
I did not put the leading '?' based on https://github.com/aspnet/HttpAbstractions/blob/dev/src/Microsoft.AspNetCore.WebUtilities/QueryHelpers.cs#L126 (and the code that follows this). I don't feel strongly about this so I can add '?' if you prefer to have it.